src/trading_api_functions.jl

```diff
@@ -0,0 +1,541 @@
+limit_price	        string<number>	            Optional                required if type is limit or stop_limit
+stop_price	        string<number>	            Optional                required if type is stop or stop_limit
+trail	            string<number>	            Optional                the new value of the trail_price or trail_percent value (works only for type=“trailing_stop”)
+client_order_id	    string (<= 48 characters)	Optional                A unique identifier for the order. Automatically generated if not sent.
+
+#  examples
+order_id_string = "1483d27e-f49b-4216-9d20-e78a0541a049"
+replace_an_order(order_id_string; qty="1",time_in_force="gtc")
+
+"""
+function replace_an_order(order_id::String; qty::Any=nothing, time_in_force::Any=nothing, limit_price::Any=nothing, stop_price::Any=nothing, trail::Any=nothing, client_order_id::Any=nothing)::DataFrame
+
+    # end point url for orders at the live or paper account
+    url = join([TRADING_API_URL, "orders"], "/")
+
+    params = Dict(
+    "qty" => qty,
+    "time_in_force" => time_in_force,
+    "limit_price" => limit_price, 
+    "stop_price" => stop_price,
+    "trail" => trail, 
+    "client_order_id" => client_order_id
+    )
+
+    # join order_id 
+    url = join([url, order_id], "/")
+    # build parameters into the url seperated by ?
+    paramsurl = params_question_mark_sep(params)
+    url = join([url, paramsurl], "?")
+    # send url and body
+    res = HTTP.patch(url, body=json(params), headers = HEADERS[])
+    resdict = JSON.parse(String(res.body))
+    # save response to dataframe
+    resdf = DataFrame(resdict)
+    print(DataFrame([[names(resdf)]; collect.(eachrow(resdf))], [:column; Symbol.(axes(resdf, 1))]))
+    return resdf
+end
+
+
+"""
+
+function get_orders_by_order_id(order_id::Any)::DataFrame
+
+GET /v2/orders/{order_id}
+Retrieves a single order for the given order_id
+
+Query Parameters
+Attribute 	Type 	          Requirement 	  Description
+order_id 	string<uuid>      Optional        Order ID
+
+#  examples
+orders_by_id_df = get_orders_by_order_id(order_id)
+orders_by_id_df = get_orders_by_order_id(nothing)
+
+
+"""
+function get_orders_by_order_id(order_id::String)::DataFrame
+
+    # end point url for orders at the live or paper account
+    url = join([TRADING_API_URL, "orders"], "/")
+
+    # order_id is optional
+    # pull all orders or per the order id
+    if 1 == isnothing(order_id)
+        # end point url for orders at the live or paper account
+        url = join([TRADING_API_URL, "orders"], "/")
+    elseif 1 != isnothing(order_id)
+        url = join([url, order_id], "/")
+    end
+
+    res = HTTP.get(url, headers = HEADERS[])
+    resdict = JSON.parse(String(res.body))
+    resdf = DataFrame(resdict)
+    print(DataFrame([[names(resdf)]; collect.(eachrow(resdf))], [:column; Symbol.(axes(resdf, 1))]))
```

Comment:
Let's not print by default, this function could be called many times and this will slow down. Let the user print the output if they need to

Reply:
changed all cases of print() - will not print by default.